runner-proxy/server/handle.go

```diff
@@ -0,0 +1,113 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/spaceuptech/helpers"
+	"github.com/spaceuptech/space-cloud/runner-proxy/utils"
+)
+
+func (s *Server) handleProxy() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(r.Context(), 30*time.Minute)
+		defer cancel()
+
+		// Close the body of the request
+		defer utils.CloseTheCloser(r.Body)
+
+		// http: Request.RequestURI can't be set in client requests.
+		// http://golang.org/src/pkg/net/http/client.go
+		r.RequestURI = ""
+
+		// Get the meta data from headers
+		project := r.Header.Get("x-og-project")
+		service := r.Header.Get("x-og-service")
+		ogHost := r.Header.Get("x-og-host")
+		ogPort := r.Header.Get("x-og-port")
+		ogVersion := r.Header.Get("x-og-version")
+
+		// Delete the headers
+		r.Header.Del("x-og-project")
+		r.Header.Del("x-og-service")
+		r.Header.Del("x-og-host")
+		r.Header.Del("x-og-port")
+		r.Header.Del("x-og-version")
+
+		// Change the destination with the original host and port
+		r.Host = ogHost
+		r.URL.Host = fmt.Sprintf("%s:%s", ogHost, ogPort)
+
+		// Set the url scheme to http
+		r.URL.Scheme = "http"
+
+		helpers.Logger.LogDebug(helpers.GetRequestID(ctx), fmt.Sprintf("Proxy is making request to host (%s) port (%s)", ogHost, ogPort), nil)
+
+		id := fmt.Sprintf("%s-%s-%s", project, service, ogVersion)
+		item, ok := s.m.m[id]
```

Comment:
```suggestion
		item, ok := s.cache.getDeployment(id)
```
Don't access variables of a struct directly, create methods